cdap-ui/app/features/adapters/controllers/create/canvas-ctrl.js

```diff
@@ -5,24 +5,24 @@ angular.module(PKG.name + '.feature.adapters')
         transformTemplates = [],
         sinkTemplates = [];
 
+    function objectToArray(obj, arr) {
+      angular.forEach(obj, function (val) {
+        if (val.templateType === MyPlumbService.metadata.template.type) {
+          val.icon = 'fa-plug';
+          val.name = val.templateName;
+
+          arr.push(val);
+        }
+      });
+    }
+
     mySettings.get('pluginTemplates')
       .then(function (res) {
-        var templates = res[$state.params.namespace];
-
-        angular.forEach(templates, function (template) {
-          if (template.templateType === MyPlumbService.metadata.template.type) {
-            template.icon = 'fa-plug';
-            template.name = template.templateName;
-            if (template.type === 'source') {
-              sourceTemplates.push(template);
-            } else if (template.type === 'transform') {
-              transformTemplates.push(template);
-            } else if (template.type === 'sink') {
-              sinkTemplates.push(template);
-            }
-          }
-        });
+        var templates = res[$state.params.namespace][MyPlumbService.metadata.template.type];
 
+        objectToArray(templates.source, sourceTemplates);
```

Comment:
I am guessing we could change here if we are changing it here (https://github.com/caskdata/cdap/pull/3655/files#r37775127). Instead of passing in a reference and have the utility function change the array we could get an array as return type and we could do whatever we want with it. Thoughts?